### PR TITLE
A behaviour change to query_mgmt_db.py --mode todo

### PR DESCRIPTION
**Pull request by sungeunbae**
`query_mgmt_db.py --mode todo` is supposed to give a list of jobs that still need to be done. I think it makes more sense if it shows everything that hasn't reached "completed" status.

### CODE REVIEW

**Review by joelridden — COMMENTED** [review]
I don't think this makes sense to me? The tasks that are todo are ones that haven't been put in the queue or touched at all, which are the ones that are set to created? Otherwise you get the ones here that failed, killed by wct, ones waiting in queue and ones currently running etc based on this SQL

**Comment by sungeunbae** [comment]
When running the whole cybershake, I seriously wanted to have a way to see what more jobs need to be completed. --mode todo only showing "created" jobs is not particularly useful in my opinion.  
Can you describe a situation where we want to see the list of untouched created jobs only?  

I still feel "todo" means jobs, regardless of the current status, that still need work

**Comment by sungeunbae** [comment]
Updated the query logic.
Without todo mode, you will see
```
...
               MS04_REL29 |          EMOD3D |     failed | 13251611 |  2023-07-19 01:12:08
               MS04_REL29 |          EMOD3D |     failed | 13336848 |  2023-08-08 21:40:48
               MS04_REL30 |          EMOD3D |     failed | 13251613 |  2023-07-19 01:12:08
               MS04_REL30 |          EMOD3D |     failed | 13336849 |  2023-08-08 21:40:48
               MS09_REL01 |          EMOD3D |  completed | 13336850 |  2023-08-08 21:40:48
               MS09_REL01 |          EMOD3D |     failed | 13251615 |  2023-07-19 01:12:08
               MS09_REL02 |          EMOD3D |  completed | 13336852 |  2023-08-08 21:40:48
               MS09_REL02 |          EMOD3D |     failed | 13251617 |  2023-07-19 01:12:08
               MS09_REL03 |          EMOD3D |  completed | 13336853 |  2023-08-08 21:40:48
               MS09_REL03 |          EMOD3D |     failed | 13251619 |  2023-07-19 01:12:08
...
               MS09_REL21 |          EMOD3D |  completed | 13336879 |  2023-08-08 21:40:48
               MS09_REL21 |          EMOD3D |     failed | 13280910 |  2023-07-27 22:53:35
               MS09_REL22 |          EMOD3D |     failed | 13280912 |  2023-07-27 22:53:35
               MS09_REL22 |          EMOD3D |     failed | 13336881 |  2023-08-08 21:40:48
               MS09_REL23 |          EMOD3D |  completed | 13336882 |  2023-08-08 21:40:48
               MS09_REL23 |          EMOD3D |     failed | 13280915 |  2023-07-27 23:04:33
...
         UpperSlope_REL20 |          EMOD3D |    created |     None |  2023-08-07 01:40:09
         UpperSlope_REL20 |          EMOD3D |     failed | 13323625 |  2023-08-06 21:14:48
         UpperSlope_REL21 |          EMOD3D |    created |     None |  2023-08-07 01:40:09
         UpperSlope_REL21 |          EMOD3D |     failed | 13323626 |  2023-08-06 21:14:48
         UpperSlope_REL22 |          EMOD3D |    created |     None |  2023-08-07 01:40:09
         UpperSlope_REL22 |          EMOD3D |     failed | 13323627 |  2023-08-06 21:14:48

```
With todo mode,
```
...
               MS04_REL29 |          EMOD3D |     failed | 13336848 |  2023-08-08 21:40:48
               MS04_REL30 |          EMOD3D |     failed | 13336849 |  2023-08-08 21:40:48
               MS09_REL22 |          EMOD3D |     failed | 13336881 |  2023-08-08 21:40:48
...
         UpperSlope_REL20 |          EMOD3D |    created |     None |  2023-08-07 01:40:09
         UpperSlope_REL21 |          EMOD3D |    created |     None |  2023-08-07 01:40:09
         UpperSlope_REL22 |          EMOD3D |    created |     None |  2023-08-07 01:40:09

```
It shows the latest non-complete status of each task - We have just one entry per (rel,process_type) combo. This should give a better overview of the what jobs remain to be done.

**Review by joelridden — DISMISSED** [review]
Does complicate a little, but I like the result and think it'd be useful. Still don't fully agree with the todo meaning but I'm fine to part with the opinion.

**Review by joelridden — DISMISSED** [review]

**Review by claudio525 — APPROVED** [review]

File: workflow/automation/lib/shared_template.py [hunk 1]
```diff
@@ -133,6 +133,10 @@ def resolve_header(
         template_path = platform_config[const.PLATFORM_CONFIG.HEADER_FILE.name]
 
     j2_env = Environment(loader=FileSystemLoader(template_dir), trim_blocks=True)
+
+    from qcore import config
+    wallclock_limit = min(wallclock_limit, str(config.qconfig[config.ConfigKeys.MAX_JOB_WCT.name]))
```
Comment (joelridden) [inline workflow/automation/lib/shared_template.py:138]: This shouldn't be needed as that should be covered by the confine_wct_node_parameters function

File: workflow/automation/lib/shared.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
 import sys
 from datetime import datetime
 from logging import Logger, DEBUG, INFO
+from pathlib import Path
+import shutil
```
Comment (joelridden) [inline workflow/automation/lib/shared.py:18]: Adding imports that are not being used?

**Pull request closed without merging**